### PR TITLE
Generate checkpoint path using TRAINING_JOB_NAME environment variable

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
This change adds to the behavior of determining a path for storing checkpoints during training. If a path has been specified through the hyperparameters provided by the Python SDK, then there is no change in behavior. If a container is being run via local mode or just locally on its own without the SDK, then there is also no change in behavior.

The only difference in behavior is if the container is being run on SageMaker but the request was submitted without using the Python SDK. In this case, the checkpoint path will be generated in a similar fashion to [how the Python SDK does it](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/tensorflow/estimator.py#L307-L308). This takes advantage of the [new environment variable](https://github.com/aws/sagemaker-containers/pull/34) available through training on SageMaker.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/tf_container/train_entry_point.py [hunk 1]
```diff
@@ -101,10 +104,22 @@ def _get_trainer_class():
     return tf_container.trainer.Trainer
 
 
+def _get_checkpoint_dir(env):
+    if 'checkpoint_path' not in env.hyperparameters:
+        return env.model_dir
```
Comment (yuanzhua) [inline src/tf_container/train_entry_point.py]: why not os.path.join(env.model_dir, job_name, 'checkpoints')

File: test/unit/test_train_entry_point.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+def test_get_checkpoint_dir_without_checkpoint_path(train_entry_point_module):
+    env = Mock(name='env', hyperparameters={}, model_dir='model/output')
+    checkpoint_dir = train_entry_point_module._get_checkpoint_dir(env)
+
+    assert checkpoint_dir == 'model/output'
+
+
+def test_get_checkpoint_dir_without_tuning(train_entry_point_module):
+    env = Mock(name='env', hyperparameters={'checkpoint_path': CHECKPOINT_PATH}, job_name=None)
```
Comment (lukmis) [inline test/unit/test_train_entry_point.py]: Please either add a test where job_name=None and checkpoint_path.endswith(os.path.join(job_name, 'checkpoints')) to cover the 'if' cases.

Comment (laurenyu) [reply test/unit/test_train_entry_point.py]: Doesn't this one cover it? if `job_name` is `None`, then `if job_name is None or checkpoint_path.endswith(os.path.join(job_name, 'checkpoints'))` will evaluate to `True` regardless of the `checkpoint_path`

The test below checks for if `checkpoint_path` already matches `'<training-job>/checkpoints'`
